compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1289,11 +1296,23 @@ impl<'hir> LoweringContext<'_, 'hir> {
         }
     }
 
-    pub(super) fn lower_generics_mut(
+    /// Return the pair of the lowered `generics` as `hir::Generics` and the evaluation of `f` with
+    /// the carried impl trait definitions and bounds.
+    fn lower_generics<T>(
         &mut self,
         generics: &Generics,
-        mut itctx: ImplTraitContext<'_, 'hir>,
-    ) -> GenericsCtor<'hir> {
+        parent_node_id: NodeId,
+        itctx: ImplTraitContext,
+        f: impl FnOnce(&mut Self) -> T,
+    ) -> (&'hir hir::Generics<'hir>, T) {
+        match itctx {
+            ImplTraitContext::Universal(..) => {}
+            _ => {
+                debug_assert!(self.impl_trait_defs.is_empty());
+                debug_assert!(self.impl_trait_bounds.is_empty());
```

Comment:
Should these always be empty when entering this function?

Reply:
You meant in the not `Universal` case or what? Unsure I got what you meant here.

Reply:
I mean: should we remove the match and always assert?